api.planx.uk/helpers.ts

```diff
@@ -83,53 +87,42 @@ const createAssociatedOperation = async (flowId: Flow["id"]) => {
     },
   );
 
-  return data?.insert_operations_one;
+  return data?.operation;
 };
 
+interface PublishedFlows {
+  flow: {
+    publishedFlows: {
+      // TODO: use FlowGraph from planx-core here
+      data: Flow["data"];
+    }[];
+  } | null;
+}
+
 // Get the most recent version of a published flow's data (flattened, with external portal nodes)
 const getMostRecentPublishedFlow = async (
   id: string,
 ): Promise<Flow["data"]> => {
-  const data = await adminClient.request(
+  const { flow } = await $public.client.request<PublishedFlows>(
     gql`
       query GetMostRecentPublishedFlow($id: uuid!) {
-        flows_by_pk(id: $id) {
-          published_flows(limit: 1, order_by: { created_at: desc }) {
-            data
-          }
-        }
-      }
-    `,
-    { id },
-  );
-
-  return data.flows_by_pk.published_flows?.[0]?.data;
-};
-
-// Get the snapshot of the published flow for a certain point in time (flattened, with external portal nodes)
-//   created_at refers to published date, value passed in as param should be lowcal_session.updated_at
-const getPublishedFlowByDate = async (id: string, created_at: string) => {
-  const data = await adminClient.request(
-    gql`
-      query GetPublishedFlowByDate($id: uuid!, $created_at: timestamptz!) {
-        flows_by_pk(id: $id) {
-          published_flows(
+        flow: flows_by_pk(id: $id) {
+          publishedFlows: published_flows(
             limit: 1
             order_by: { created_at: desc }
-            where: { created_at: { _lte: $created_at } }
           ) {
             data
           }
         }
       }
     `,
-    {
-      id,
-      created_at,
-    },
+    { id },
   );
 
-  return data.flows_by_pk.published_flows?.[0]?.data;
+  const mostRecent = flow?.publishedFlows?.[0]?.data;
+  if (!mostRecent) throw Error(`Published flow not found for flow ${id}`);
```

Comment:
Generally we do error catching outside these little request functions but making it once here actually is much cleaner.

Going forward to would be nice to handle this in a much more standardised method - I actually think Apollo has some really nice patterns for this which may be something to consider.